IDEAS/Examples/IBPSA/TwoWayOverwriteTRV.mo

```diff
@@ -0,0 +1,231 @@
+  parameter Modelica.Blocks.Types.Init init=Modelica.Blocks.Types.Init.InitialOutput
+    "Type of initialization (no init/steady state/initial state/initial output)"
+    annotation(Dialog(tab="Dynamics", group="Filtered opening",enable=use_inputFilter));
+  parameter Real y_start=1 "Initial value of control signal"
+    annotation(Dialog(tab="Dynamics", group="Filtered opening",enable=use_inputFilter));
+  parameter Modelica.SIunits.PressureDifference dpFixed_nominal(displayUnit="Pa", min=0) = 0
+    "Pressure drop of pipe and other resistances that are in series"
+     annotation(Dialog(group = "Nominal condition"));
+
+  parameter Real l(min=1e-10, max=1) = 0.0001
+    "Valve leakage, l=Kv(y=0)/Kv(y=1)";
+
+  parameter Boolean from_dp = false
+    "= true, use m_flow = f(dp) else dp = f(m_flow)"
+    annotation (Evaluate=true, Dialog(tab="Advanced"));
+
+  final parameter Modelica.SIunits.PressureDifference dp_nominal(displayUnit="Pa")=dpValve_nominal + dpFixed_nominal
+    "Pressure drop at nominal mass flow rate"
+    annotation(Dialog(group = "Nominal condition"));
+  parameter Boolean homotopyInitialization = true "= true, use homotopy method"
+    annotation(Evaluate=true, Dialog(tab="Advanced"));
+  parameter Boolean linearized = false
+    "= true, use linear relation between m_flow and dp for any flow rate"
+    annotation(Evaluate=true, Dialog(tab="Advanced"));
+
+  final parameter Modelica.SIunits.MassFlowRate m_flow_turbulent(min=0)=deltaM * abs(m_flow_nominal)
+    "Turbulent flow if |m_flow| >= m_flow_turbulent";
+
+  IDEAS.Fluid.Actuators.Valves.TwoWayLinear val(
+    redeclare package Medium = Medium,
+    m_flow_nominal=m_flow_nominal,
+    dpValve_nominal=dpValve_nominal,
+    allowFlowReversal=allowFlowReversal,
+    show_T=show_T,
+    from_dp=from_dp,
+    homotopyInitialization=homotopyInitialization,
+    linearized=linearized,
+    deltaM=deltaM,
+    rhoStd=rhoStd,
+    use_inputFilter=use_inputFilter,
+    riseTime=riseTime,
+    init=init,
+    y_start=y_start,
+    dpFixed_nominal=dpFixed_nominal,
+    l=l,
+    CvData=IDEAS.Fluid.Types.CvTypes.OpPoint) "Linear valve"
+    annotation (Placement(transformation(extent={{-10,-10},{10,10}})));
+
+  Modelica.Blocks.Interfaces.RealInput T(unit="K") "Temperature measurement" annotation (
+      Placement(transformation(
+        extent={{-20,-20},{20,20}},
+        rotation=270,
+        origin={0,106})));
+  Modelica.Blocks.Sources.RealExpression yExp(y=
+        IDEAS.Utilities.Math.Functions.smoothLimit(
+        (TSet - T)/P,
+        0,
+        1,
+        0.001))        "Control signal"
+    annotation (Placement(transformation(extent={{-94,30},{-54,50}})));
+  Modelica.Blocks.Interfaces.RealOutput y "Valve set point"
+    annotation (Placement(transformation(extent={{40,60},{60,80}}),
+        iconTransformation(extent={{40,60},{60,80}})));
+
+  IBPSA.Utilities.IO.SignalExchange.Overwrite oveVal(description=
```

Comment:
Please use IDEAS components only in IDEAS.

Reply:
https://github.com/open-ideas/IDEAS/pull/1111/commits/8795163e7adcff1dbef3bd7ef6fa9faf2b9ab412